src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -1,19 +1,209 @@
 };
+use tokio::sync::mpsc;
+use tracing::{error, info};
 
 use super::{
     check_ttl, conv_u64_sec_2_i64, fmt, DeleteOption, KVEngine, KeyType, KvVersion, LockKeyType,
     MetaTxn, SetOption, ValueType,
 };
 use crate::common::error::{Context, DatenLordResult};
 
+/// The keepalive session struct
+#[derive(Clone)]
+pub struct Session {
+    close_tx: mpsc::Sender<()>,
+    inner: Arc<SessionInner>,
+}
+
+impl Debug for Session {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Session").finish()
+    }
+}
+
+impl Session {
+    /// Create a new session
+    pub fn new(client: etcd_client::Client, ttl: i64, lease_id: i64) -> Arc<Self> {
+        let is_closed = Arc::new(AtomicBool::new(false));
+        let inner = Arc::new(SessionInner {
+            is_closed,
+            lease_id,
+            ttl,
+            client,
+        });
+        let (close_tx, close_rx) = mpsc::channel(1);
+
+        let _inner = inner.clone();
+        tokio::spawn(async move {
+            _inner.keep_alive(close_rx).await.unwrap();
+        });
+
+        let session = Session { close_tx, inner };
+
+        return Arc::new(session);
+    }
+
+    /// Get the lease id
+    pub fn lease_id(&self) -> i64 {
+        self.inner.lease_id
+    }
+
+    /// Get the ttl
+    pub fn ttl(&self) -> i64 {
+        self.inner.ttl
+    }
+
+    /// Get the is_closed flag
+    pub fn is_closed(&self) -> bool {
+        self.inner
+            .is_closed
+            .load(std::sync::atomic::Ordering::Relaxed)
+    }
+
+    /// Get the is_closed
+    pub fn close(&self) {
+        let _ = self.close_tx.try_send(());
+        self.inner
+            .close();
+    }
+}
+
+impl Drop for Session {
+    fn drop(&mut self) {
+        // Set the is_closed flag
+        self.close()
+    }
+}
+
+/// The keepalive session inner struct
+pub struct SessionInner {
+    is_closed: Arc<AtomicBool>,
+    lease_id: i64,
+    ttl: i64,
+    client: etcd_client::Client,
+}
+
+impl Debug for SessionInner {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SessionInner")
+            .field("is_closed", &self.is_closed)
+            .field("lease_id", &self.lease_id)
+            .field("ttl", &self.ttl)
+            .finish()
+    }
+}
+
+impl SessionInner {
+    /// Get the is_closed flag
+    fn is_closed(&self) -> bool {
+        self.is_closed
+            .load(std::sync::atomic::Ordering::Relaxed)
+    }
+
+    /// Set close flag
+    fn close(&self) {
+        self.is_closed
+            .store(true, std::sync::atomic::Ordering::Relaxed);
+    }
+
+    /// Reopen close flag
+    fn reopen(&self) {
+        self.is_closed
+            .store(false, std::sync::atomic::Ordering::Relaxed);
+    }
+
+    /// Keep alive the lease
+    pub async fn keep_alive(&self, mut close_rx: mpsc::Receiver<()>) -> DatenLordResult<()> {
```

Comment:
It seems that an error would never be returned from this method.

Reply:
It's a loop function, we will try to keep it as daemon task, so we don't return an error directly.

Reply:
Remove the result function.

Reply:
Yes, just return `()` insteads of a `Result`.